crates/symbolicator/src/services/cacher.rs

```diff
@@ -277,51 +295,93 @@ impl<T: CacheItemRequest> Cacher<T> {
-            sentry::configure_scope(|scope| {
-                scope.set_extra(
-                    &format!("cache.{}.cache_path", name),
-                    cache_path.to_string_lossy().into(),
-                );
-            });
+        let status = status.unwrap_or_else(|| CacheStatus::from_content(&byte_view));
 
-            log::trace!("Creating {} at path {:?}", name, cache_path);
-        }
+        let path = match self.config.cache_dir() {
+            Some(ref cache_dir) => {
+                // Cache is enabled, write it!
+                let cache_path = key.cache_path(cache_dir, T::VERSIONS.current);
 
-        let byteview = ByteView::open(temp_file.path())?;
-
-        metric!(
-            counter(&format!("caches.{}.file.write", name)) += 1,
-            "status" => status.as_ref(),
-            "is_refresh" => &is_refresh.to_string(),
-        );
-        metric!(
-            time_raw(&format!("caches.{}.file.size", name)) = byteview.len() as u64,
-            "hit" => "false",
-            "is_refresh" => &is_refresh.to_string(),
-        );
-
-        let path = match cache_path {
-            Some(ref cache_path) => {
-                status.persist_item(cache_path, temp_file)?;
-                CachePath::Cached(cache_path.to_path_buf())
+                log::trace!(
+                    "Creating {} at path {:?}",
+                    self.config.name(),
+                    cache_path.display()
+                );
+                let parent = cache_path.parent().ok_or_else(|| {
+                    std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        "no parent directory to persist item",
+                    )
+                })?;
+                fs::create_dir_all(parent).await?;
+                let mut file = temp_file.persist(&cache_path).map_err(|x| x.error)?;
+
+                let file_size = file.seek(SeekFrom::End(0))?;
+                metric!(
+                    counter(&format!("caches.{}.file.write", self.config.name())) += 1,
+                    "status" => status.as_ref(),
+                    "is_refresh" => &is_refresh.to_string(),
+                );
+                metric!(
+                    time_raw(&format!("caches.{}.file.size", self.config.name())) = file_size,
+                    "hit" => "false",
+                    "is_refresh" => &is_refresh.to_string(),
+                );
+                sentry::configure_scope(|scope| {
+                    scope.set_extra(
+                        &format!("cache.{}.cache_path", self.config.name()),
+                        cache_path.to_string_lossy().into(),
+                    );
+                });
+
+                CachePath::Cached(cache_path)
             }
             None => CachePath::Temp(temp_file.into_temp_path()),
         };
 
-        Ok(request.load(key.scope.clone(), status, byteview, path))
+        let byte_view = ByteView::open(&path)?;
```

Comment:
with all the kerfuffle around `ByteView::map_file_ref`, you are still re-opening the `byte_view` here. I guess this is just an oversight?

Reply:
yes it was